OneCommunityGlobal/HighestGoodNetworkApp
# Shihao - Improve Summary Loading Performance

# Description
![Screenshot 2023-08-24 at 17 51 20](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/48acf93f-bc50-4a9c-b3df-745561cee84f)


## Related PRS (if any):
This frontend PR is related to the [#484](https://github.com/OneCommunityGlobal/HGNRest/pull/484) backend PR.

## Main changes explained:
- Refactor `FormattedReport` component
- Adjust query to reduce HTTP request
- Add an observer to check if the dom is visible

## How to test:
1. check into `development` first
2. log as owner user
3. go to dashboard→ Reports→ Weekly Summaries Report
4. wait for page fully loaded


1. then check into current branch
6. log as owner user
7. go to dashboard→ Reports→ Weekly Summaries Report
8. check if the page load faster than dev branch

## Screenshots or videos of changes:
**Dev branch**

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/422d6e50-7070-4294-a1c3-692c0960f6a9

**After changes**

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/0e6aca47-9f33-4ca9-a12f-27e541f825ab
